sql/sqlite/driver.go

```diff
@@ -566,6 +566,77 @@ func autoinc(t *schema.Table) {
 	t.PrimaryKey.Parts[0].C.Attrs = append(t.PrimaryKey.Parts[0].C.Attrs, AutoIncrement{})
 }
 
+// The following regexes extract named foreign-key constraints defined in the table-constraints or inlined
+// as column-constraints. Note, we assume the SQL statements are valid as they are returned by SQLite.
+var (
+	reConstC = regexp.MustCompile("(?i)(?:[(,]\\s*)([\"`\\w]+).*\\s+CONSTRAINT\\s+([\"`\\w]+)\\s+REFERENCES\\s+([\"`\\w]+)\\s*\\(([,\"` \\w]+)\\)")
+	reConstT = regexp.MustCompile("(?i)CONSTRAINT\\s+[\"`]*(\\w+)[\"`]*\\s+FOREIGN\\s+KEY\\s+\\(([,\"` \\w]+)\\)\\s+REFERENCES\\s+[\"`]*(\\w+)[\"`]*\\s*\\(([,\"` \\w]+)\\)")
+)
+
+// fillConstName fills foreign-key constrain names from CREATE TABLE statement.
+func fillConstName(t *schema.Table) error {
+	var c CreateStmt
+	if !sqlx.Has(t.Attrs, &c) {
+		return fmt.Errorf("missing CREATE statment for table: %q", t.Name)
+	}
+	// Loop over table constraints.
+	for _, m := range reConstT.FindAllStringSubmatch(c.S, -1) {
+		if len(m) != 5 {
+			return fmt.Errorf("unexpected number of matches for a table constraint: %q", m)
+		}
+		// Pattern matches "constraint_name", "columns", "ref_table" and "ref_columns".
+		for _, fk := range t.ForeignKeys {
+			// Found a foreign-key match for the constraint.
+			if matchFK(fk, columns(m[2]), m[3], columns(m[4])) {
+				fk.Symbol = m[1]
+				break
+			}
+		}
+	}
+	// Loop over inlined column constraints.
+	for _, m := range reConstC.FindAllStringSubmatch(c.S, -1) {
+		if len(m) != 5 {
+			return fmt.Errorf("unexpected number of matches for a column constraint: %q", m)
+		}
+		// Pattern matches "column", "constraint_name", "ref_table" and "ref_columns".
+		for _, fk := range t.ForeignKeys {
+			// Found a foreign-key match for the constraint.
+			if matchFK(fk, columns(m[1]), m[3], columns(m[4])) {
```

Comment:
I would add 
You need to document what returns from the regex, this code starts from a hard regex and then m[part] is used along with the function.